SchedulearnBackend/SchedulearnBackend/Services/SuggestionService.cs

```diff
@@ -38,5 +38,26 @@ public async Task<List<Suggestion>> GetSuggestionsForSuggesteeAsync(int userId)
                 .ToListAsync();
         }
 
+        public async Task<Suggestion> AddNewSuggestionAsync(CreateNewSuggestion suggestionToCreate)
+        {
+            var topic = await _schedulearnContext.Topics.FindAsync(suggestionToCreate.TopicId);
+            var suggester = await _schedulearnContext.Users.FindAsync(suggestionToCreate.SuggesterId);
+            var suggestee = await _schedulearnContext.Users.FindAsync(suggestionToCreate.SuggesteeId);
+
+            if (topic == null)
+                throw new NotFoundException(Error_TopicNotFound.ReplaceArgs(suggestionToCreate.TopicId));
+            if (suggester == null)
+                throw new NotFoundException(Error_TopicNotFound.ReplaceArgs(suggestionToCreate.SuggesterId));
```

Comment:
This should be "UserNotFound" not "TopicNotFound" since you are checking if user exists